vms/platformvm/block/executor/verifier.go

```diff
@@ -243,7 +243,7 @@ func (v *verifier) ApricotAtomicBlock(b *block.ApricotAtomicBlock) error {
 		return err
 	}
 
-	v.Mempool.Remove(b.Tx)
+	v.Mempool.Remove(b.Tx.ID())
```

Comment:
A bit of weird previous behavior changed int his PR, where we can now remove a tx by its id instead of by the entire tx type. This was because we didn't maintain a mapping of txs -> their sizes, so we just required callers to know the tx they wanted to remove so we could update our metrics correctly.

Reply:
We synced offline - the reason for this is because if we accept a tx we've never seen before, we need to remove its conflicts from the mempool by removing the tx (even though we never added it to begin with). We think a better way of doing this is to just support separate apis for removal by tx id and remove conflicts by inputs.